doc/index.rst

```diff
@@ -9,100 +7,176 @@
         :padding: 2
+        :columns: 12
 
-        Understand how Matplotlib works
+        **Reference**
         ^^^
 
-        - :ref:`users-guide-explain` in the :doc:`Users guide
-          <users/index>` has a number of advanced topics.
+        .. grid:: 1 1 2 2
+            :class-row: sd-align-minor-center
 
-    .. grid-item-card::
-        :padding: 2
+            .. grid-item::
 
-        Reference
-        ^^^
+                .. toctree::
+                    :maxdepth: 1
 
-        - :doc:`API Reference <api/index>`
-        - :doc:`Axes API <api/axes_api>` for most plotting methods
-        - :doc:`Figure API <api/figure_api>` for figure-level methods
-        - Top-level interfaces to create:
+                    API reference <api/index>
+                    Figure methods <api/figure_api>
+                    Plotting methods <api/axes_api>
+
+
+            .. grid-item::
+
+                Top-level interfaces to make:
+
+                - figures: `.pyplot.figure`
+                - subplots: `.pyplot.subplots`, `.pyplot.subplot_mosaic`
+
+
+Ecosystem
+=========
+
+.. grid:: 1 1 2 2
+    :class-row: sd-align-minor-center
+
+    .. grid-item::
+
+        .. toctree::
+            :maxdepth: 2
+
+            users/resources/index.rst
+
+    .. grid-item::
 
-          - Figures (`.pyplot.figure`)
-          - Subplots (`.pyplot.subplots`, `.pyplot.subplot_mosaic`)
+        :octicon:`link-external;1em;sd-text-info` `Third-party packages <https://matplotlib.org/mpl-third-party/>`_,
 
+        provide custom, domain specific, and experimental features, including
+        styles, colors, more plot types and backends, and alternative
+        interfaces.
 
-********************
-Third-party packages
-********************
 
-There are many `Third-party packages
-<https://matplotlib.org/mpl-third-party/>`_ built on top of and extending
-Matplotlib.
 
 
-************
-Contributing
-************
+About
+=====
+
+.. grid:: 1 1 2 2
+    :class-row: sd-align-minor-center
+
+    .. grid-item::
+
+        .. raw:: html
+
+            <div class="grid__intro" id="image_rotator"></div>
+
+    .. grid-item::
+
+        .. toctree::
+            :maxdepth: 2
+
+            users/project/index.rst
+
+
+Contribute
+==========
+
+.. grid:: 1 1 2 2
+    :class-row: sd-align-minor-center
+
+    .. grid-item::
+
+        Matplotlib is a community project maintained for and by
+        its users. There are many ways you can help!
+
+    .. grid-item::
+        .. toctree::
+            :maxdepth: 2
+
+            devel/index.rst
+
+
+Releases
+========
+
+.. grid:: 1 1 2 2
+    :class-row: sd-align-minor-center
+
+    .. grid-item::
+        New versions are generally released every 6-8 months.
+        New features and API changes are described in the release notes.
+
+    .. grid-item::
 
-Matplotlib is a community project maintained for and by its users.  There are many ways
-you can help!
+        .. toctree::
+            :maxdepth: 2
```

Comment:
```suggestion
            :maxdepth: 1
```

The toc looks a bit too large compared to the description block on the left, and I don't think direct click-through links are needed here. Checking release notes is not something one does often. And anyway it may be simpler to click the "Releases" navbar item rather than scrolling down. So this is here manily for structure and as an entry point but not so much for quick navigation.

![image](https://github.com/matplotlib/matplotlib/assets/2836374/c8855cb7-67b9-4925-9a2e-5e9671784651)